### PR TITLE
feat!: differentiate swarms based on network_id

### PR DESCRIPTION
**Pull request by Ma233**
### Please check if the PR fulfills these requirements

- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## :large_blue_circle: What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
This PR introduces a `network_id` field in `ConnectNodeSend` message. It's used to prevent nodes from different networks from connecting to each other.

## :radioactive: Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
There is a breaking change in config file.
User must specify network_id in config file now.

## :information_source: Other information

Closes #issue

### CODE REVIEW

File: crates/core/src/message/handlers/connection.rs [hunk 1]
```diff
@@ -60,6 +60,10 @@ impl HandleMsg<QueryForTopoInfoReport> for MessageHandler {
 #[cfg_attr(not(feature = "wasm"), async_trait)]
 impl HandleMsg<ConnectNodeSend> for MessageHandler {
     async fn handle(&self, ctx: &MessagePayload, msg: &ConnectNodeSend) -> Result<()> {
+        if msg.network_id != self.transport.network_id {
+            return Ok(());
```
Comment (Ma233) [inline crates/core/src/message/handlers/connection.rs:64]: Should the current node directly not respond when receiving a handshake request from a different network node?

Comment (RyanKung) [reply crates/core/src/message/handlers/connection.rs:64]: Yes, response of denial of connection may be expensive in some case, especially the signature/verify phase.

**Comment by Ma233** [comment]
@RyanKung Hi, it's a breaking change in config file. We may should upgrade the middle version number.

**Review by RyanKung — APPROVED** [review]

**Comment by RyanKung** [comment]
> @RyanKung Hi, it's a breaking change in config file. We may should upgrade the minor version number.

Yep, version number will be upgrade on PR https://github.com/RingsNetwork/rings/pull/578

**Pull request merged**